spyder/widgets/reporterror.py

```diff
@@ -265,70 +356,57 @@ def set_require_minimum_length(self, state):
-                # directly
-                main = self.parent()
-        else:
-            main = None
-
         # Getting description and traceback
         title = self.title.text()
         description = self.input_description.toPlainText()
         traceback = self.error_traceback[:-1]  # Remove last EOL
 
         # Render issue
-        if main is not None:
-            issue_text = main.render_issue(description=description,
+        if self.is_report:
+            issue_text = self.render_issue(description=description,
                                            traceback=traceback)
         else:
             issue_text = description
 
         try:
-            if main is None:
-                if self._github_org == 'spyder-ide':
-                    org = 'ccordoba12'
+            if running_under_pytest():
+                org = 'ccordoba12'
             else:
                 org = self._github_org
 
             repo = self._github_repo
-            github_backend = GithubBackend(org, repo, parent_widget=main)
+            github_backend = GithubBackend(org, repo, parent_widget=self)
             github_report = github_backend.send_report(title, issue_text)
+
             if github_report:
                 self.close()
-
         except Exception:
             ret = QMessageBox.question(
-                      self, _('Error'),
-                      _("An error occurred while trying to send the issue to "
-                        "Github automatically. Would you like to open it "
-                        "manually?<br><br>"
-                        "If so, please make sure to paste your clipboard "
-                        "into the issue report box that will appear in a new "
-                        "browser tab before clicking <i>Submit</i> on that "
-                        "page."))
+                self,
+                _('Error'),
+                _("An error occurred while trying to send the issue to "
+                  "Github automatically. Would you like to open it "
+                  "manually?<br><br>"
+                  "If so, please make sure to paste your clipboard "
+                  "into the issue report box that will appear in a new "
+                  "browser tab before clicking <i>Submit</i> on that "
+                  "page."),
+            )
+
             if ret in [QMessageBox.Yes, QMessageBox.Ok]:
                 QApplication.clipboard().setText(issue_text)
                 issue_body = (
                     " \n<!---   *** BEFORE SUBMITTING: PASTE CLIPBOARD HERE "
                     "TO COMPLETE YOUR REPORT ***   ---!>\n")
-                if main is not None:
-                    main.report_issue(body=issue_body, title=title,
-                                      open_webpage=True)
-                else:
-                    pass
+
+                if self.is_report:
+                    self.open_web_report(body=issue_body, title=title)
```

Comment:
I think this is incorrect because we need to open a webpage with the issue when GH authentication fails, no matter if the issue was generated by a user report or an internal traceback.

Besides, I don't understand why you're associating `if main is not None` with `if self.is_report` (you did the same above).

Reply:
Fixed